tests/support/client.rs

```diff
@@ -8,43 +8,57 @@ use self::tokio::{
     io::{AsyncRead, AsyncWrite},
     net::TcpStream,
 };
+use bytes::{Buf, BufMut};
+use rustls::{ClientConfig, ClientSession};
+use std::io::{Read, Write};
+use std::sync::Arc;
 use support::bytes::IntoBuf;
 use support::hyper::body::Payload;
+use tokio_rustls::TlsStream;
+use webpki::DNSName;
 
+type ClientError = hyper::Error;
 type Request = http::Request<Bytes>;
 type Response = http::Response<BytesBody>;
-type Sender = mpsc::UnboundedSender<(Request, oneshot::Sender<Result<Response, String>>)>;
+type Sender = mpsc::UnboundedSender<(Request, oneshot::Sender<Result<Response, ClientError>>)>;
+type TlsConfig = (Arc<ClientConfig>, DNSName);
```

Comment:
Take it or leave it, but I'd prefer if this was a `struct` with named fields, rather than a type alias for a tuple. I think the code would be easier to read in that case.